browser/chat/chat_tab_helper.h

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_CHAT_CHAT_TAB_HELPER_H_
+#define BRAVE_BROWSER_CHAT_CHAT_TAB_HELPER_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "brave/components/chat_ui/common/chat_ui.mojom.h"
+#include "content/public/browser/web_contents_observer.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+namespace ui {
+class AXTree;
+}  // namespace ui
+
+using chat_ui::mojom::CharacterType;
+using chat_ui::mojom::ConversationTurn;
+
+class ChatTabHelper : public content::WebContentsObserver,
+                      public content::WebContentsUserData<ChatTabHelper> {
+  using OnArticleSummaryCallback =
+      base::OnceCallback<void(const std::u16string& summary,
+                              bool is_from_cache)>;
+
+ public:
+  class Observer : public base::CheckedObserver {
+   public:
+    ~Observer() override = default;
+
+    virtual void OnPageChanged() {}
+  };
+
+  ChatTabHelper(const ChatTabHelper&) = delete;
+  ChatTabHelper& operator=(const ChatTabHelper&) = delete;
+  ~ChatTabHelper() override;
+
+  std::string GetConversationHistoryAsString();
+  std::vector<ConversationTurn> GetConversationHistory();
```

Comment:
Return `const const std::vector<ConversationTurn>&` could let us avoid copying the entire vector here. Though currently we have to copy/transform from the unique callsite, I think copying should be done there, when it's really needed.

```suggestion
  std::vector<ConversationTurn> GetConversationHistory();
```

Reply:
The suggested change looks strange as I don't see any changes. Perhaps a GH bug?

I have change changed this to `const T&`, makes sense.